components/supervisor/frontend/src/ide/heart-beat.ts

```diff
@@ -24,7 +26,7 @@ export function schedule(wsInfo: WorkspaceInfo, sessionId: string): void {
     toCancel = new DisposableCollection()
     const sendHeartBeat = async (wasClosed?: true) => {
         try {
-            await window.gitpod.service.server.sendHeartBeat({ instanceId: wsInfo.latestInstance!.id, wasClosed });
+            window.gitpod.service.server.sendHeartBeat({ instanceId: wsInfo.latestInstance!.id, wasClosed });
```

Comment:
why should we change this line? try-catcth does not make sense anymore, and ide_close_signal can be fired even if sendHeartBeat failed?

Reply:
Ahh, I should move .catch to `sendHeartBeat` function, I thought we don't have to send `ide_close_signal` after `sendHeartBeat` is done. 

And it's another reason the previous implement unload did not works, the browser only gives around one second, `await` makes the `ide_close_signal` tracking more missable